src/RBACHandler.ts

```diff
@@ -19,20 +28,59 @@ export class RBACHandler implements Authorization {
 
     private readonly rules: RBACConfig;
 
-    constructor(rules: RBACConfig) {
+    private readonly fhirVersion: FhirVersion;
+
+    constructor(rules: RBACConfig, fhirVersion: FhirVersion) {
         this.rules = rules;
         if (this.rules.version !== this.version) {
             throw Error('Configuration version does not match handler version');
         }
+        this.fhirVersion = fhirVersion;
     }
 
     isAuthorized(request: AuthorizationRequest): boolean {
         const decoded = decode(request.accessToken, { json: true }) || {};
         const groups: string[] = decoded['cognito:groups'] || [];
 
+        if (request.exportType) {
+            return this.isExportAllowed(groups, request.exportType);
+        }
         return this.isAllowed(groups, request.operation, request.resourceType);
     }
 
+    private isExportAllowed(groups: string[], exportType: ExportType): boolean {
+        for (let index = 0; index < groups.length; index += 1) {
+            const group: string = groups[index];
+            if (this.rules.groupRules[group]) {
+                const rule: Rule = this.rules.groupRules[group];
+                if (exportType && rule.operations.includes('read')) {
+                    if (exportType === 'system') {
+                        if (
+                            (this.fhirVersion === '4.0.1' &&
+                                isEqual(rule.resources.sort(), BASE_R4_RESOURCES.sort())) ||
+                            (this.fhirVersion === '3.0.1' && isEqual(rule.resources.sort(), BASE_STU3_RESOURCES.sort()))
+                        ) {
+                            return true;
+                        }
+                    }
+                    if (exportType === 'group' || exportType === 'patient') {

```

Comment:
group and patient are the same?

Reply:
Yep, they both use Patient Compartments. A group is a "group of patients".